File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        longest = strs[0]
        

        for s in strs[1:]:
            i=0
            if s == "":
                return ""
            while i < len(longest) and i < len(s):
                if s[i] != longest[i]:
                    break
                i+=1
            longest = longest[:i]
        return longest

             

File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        
        Map<String, List<String>> result = new HashMap<>();
        for (String str : strs){
            char[] charArray = str.toCharArray();
            Arrays.sort(charArray);
            String sortedStr = new String(charArray);
            result.putIfAbsent(sortedStr,new ArrayList<>());
            result.get(sortedStr).add(str);
        }
        return new ArrayList<>(result.values());
    }
}
